modules/format/main/index.coffee

```diff
@@ -1,62 +1,78 @@
 
 siSuffixes = ['y','z','a','f','p','n','µ','','','K','M','G','T','P','E','Z','Y']
 
-zeroPad = (number, pad) ->
+zeroPad = (number, { length }) ->
   str = number.toString()
-  if str.length < pad
-    zeros = pad - str.length
+  if str.length < length
+    zeros = length - str.length
     ('0' for _ in [0..zeros-1]).join('') + str
   else str
 
-precision = (n) -> if n then Math.floor(Math.log(Math.abs(n)) / Math.LN10) else 1
+precision = (number) -> if number then Math.floor(Math.log(Math.abs(number)) / Math.LN10) else 1
 
-roundPrecision = (n, base, factor) ->
+roundPrecision = (number, base, factor) ->
   # have to deal with the +ve and -ve cases separately so that numerical errors are less likely
   if factor >= 0
-    Math.round(n / Math.pow(base, factor)) * Math.pow(base, factor)
+    Math.round(number / Math.pow(base, factor)) * Math.pow(base, factor)
   else
-    Math.round(n * Math.pow(base, - factor)) / Math.pow(base, - factor)
+    Math.round(number * Math.pow(base, - factor)) / Math.pow(base, - factor)
 
 # takes a number, and the number of significant numbers
-formatRound = (n, sf) ->
-  if isNaN(n) then return 'NaN'
-  factor = precision(n) - sf + 1
-  roundPrecision(n, 10, factor).toString()
+formatRound = (number, options) ->
+  factor = precision(number) - options.sf + 1
+  roundPrecision(number, 10, factor).toString()
 
-formatSI = (n, sf) ->
-  if isNaN(n) then return 'NaN'
-  p = Math.min(precision(n), 26) # up to yotta is supported (hence the 26)
+formatSI = (number, options) ->
+  p = Math.min(precision(number), 26) # up to yotta is supported (hence the 26)
 
   suffix = siSuffixes[Math.min(Math.max(0, Math.floor(8 + p / 3)), 16)]
 
   # calculate the scaling factor to divide by, so that the number makes sense with its si suffix
-  x = if Math.abs(n) < 1 and p % 3 and not ( -3 < p < 0) then 1000 else 1
+  x = if Math.abs(number) < 1 and p % 3 and not ( -3 < p < 0) then 1000 else 1
   if p is -3
     x = 1000
     suffix = siSuffixes[6]
   siFactor = Math.pow(10, p - p % 3) / x
 
+  formatRound(number / siFactor, options) + suffix
 
-  formatRound(n / siFactor, sf) + suffix
+formatExp = (number, options) ->
+  p = precision(number)
+  formatRound(number / Math.pow(10, p), options) + 'e' + p
 
-formatExp = (n, sf) ->
-  if isNaN(n) then return 'NaN'
-  p = precision(n)
-  formatRound(n / Math.pow(10, p), sf) + 'e' + p
+formatFixed = (number, options) -> number.toFixed(options.digits)
+
+strictCheck = (number, formatFn, options) ->
+  if options.strict
+    if isNaN(number) then 'NaN' else formatFn(number, options)
+  else
+    if hx.isString(number) then number else formatFn(number, options)
 
-formatFixed = (n, digits) ->
-  if isNaN(n) then return 'NaN'
-  n.toFixed(digits)
 
-strictCheck = (f, sf, strict) ->
-    if strict
-      (n) -> f(n, sf)
-    else
-      (n) -> if hx.isString(n) then n else f(n, sf)
+hx.round = (number, options) -> strictCheck(number, formatRound, options)
```

Comment:
The options objects should have defaults - if options is undefined, these functions will throw an error. Or if the options are required, a better error should be thrown when the options are not provided

Reply:
Decided to go with a default of 2 for the digits/sf/length.